openshift/odo
# Rename install.sh to installer.sh and Add Uninstall

## What is the purpose of this change? What does it change?
This pull request edits the `install.sh` os-independent script for macOS/Linux for installing `odo` to also uninstall `odo`. Since the script now handles both installation and uninstall, the script has been renamed `installer.sh`. 

The assumption made by these changes is that the user will have installed `odo` via this script as well.

To install `odo`, users will be able to continue using the command in the `README` for `odo`, but will need to change the name to `installer.sh`: 

```
curl -L https://github.com/openshift/odo/raw/master/scripts/installer.sh | bash
```

To uninstall `odo`, users can run the following command:

```
curl -L https://github.com/openshift/odo/raw/master/scripts/installer.sh --uninstall | bash
```

Currently, the script would only have the ability to support one flag (i.e., `--uninstall`). Ideally, the script should be able to handle multiple flags, but I'm not sure of the best way to do this in bash, so I welcome any suggestions with regard to that. 

Assuming this is an acceptable approach, I can edit the `README` to include uninstall instructions.

## Was the change discussed in an issue?
fixes #1733 

## How to test changes?
1. Run `installer.sh` on the supported operating systems (i.e., macOS and Linux distributions) to install `odo`. Verify `odo` has been installed correctly.
2. Run `installer.sh --uninstall` on the supported operating systems (i.e., macOS and Linux distributions) after installation of `odo` and verify `odo` has been properly uninstalled. 

I have also edited the tests associated with this script to test for uninstalling `odo`. Because of the name change to `installer.sh`, I also changed the test script to `test-installer.sh`.
